File: go/internal/message/message.go
package message

import (
	"encoding/json"
	"time"
)

// Event — структура сообщения, которое продюсер отправляет в топик.
type Event struct {
	ID        int       `json:"id"`
	Text      string    `json:"text"`
	Source    string    `json:"source"`
	CreatedAt time.Time `json:"created_at"`
}

// Serialize сериализует Event в JSON-байты перед отправкой в Kafka.
func Serialize(e Event) ([]byte, error) {
	return json.Marshal(e)
}

// Deserialize десериализует JSON-байты из Kafka обратно в Event.
func Deserialize(data []byte) (Event, error) {
	var e Event
	err := json.Unmarshal(data, &e)
	return e, err
}

File: anetishin/python/message.py
import json
from dataclasses import dataclass, asdict
from datetime import datetime, timezone


@dataclass
class Event:
    id: int
    text: str
    source: str
    created_at: str  # ISO 8601

    @staticmethod
    def new(id: int) -> "Event":
        return Event(
            id=id,
            text=f"message-{id}",
            source="producer",
            created_at=datetime.now(timezone.utc).isoformat(),
        )

    def serialize(self) -> bytes:
        return json.dumps(asdict(self)).encode("utf-8")

    @staticmethod
    def deserialize(data: bytes) -> "Event":
        d = json.loads(data.decode("utf-8"))
        return Event(**d)
